plugins/woocommerce/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationController.php

```diff
@@ -71,9 +72,101 @@ public function get_migrated_meta_keys() {
 	public function migrate_orders( array $order_post_ids ): void {
 		$this->error_logger = WC()->call_function( 'wc_get_logger' );
 
+		$using_transactions = $this->maybe_start_transaction();
+		if ( null === $using_transactions ) {
+			return;
+		}
+
+		$errors_were_logged = false;
+
 		foreach ( $this->all_migrators as $migrator ) {
-			$this->do_orders_migration_step( $migrator, $order_post_ids );
+			$errors_were_logged = $this->do_orders_migration_step( $migrator, $order_post_ids );
+			if ( $errors_were_logged && $using_transactions ) {
+				$this->rollback_transaction();
+				break;
+			}
+		}
+
+		if ( ! $errors_were_logged && $using_transactions ) {
+			$this->commit_transaction();
+		}
+	}
+
+	/**
+	 * Start a database transaction if the configuration mandates so.
+	 *
+	 * @return bool|null True if transaction started, false if transactions won't be used, null if transaction failed to start.
+	 */
+	private function maybe_start_transaction(): ?bool {
+		if ( 'yes' !== get_option( CustomOrdersTableController::USE_DB_TRANSACTIONS_OPTION ) ) {
+			return false;
+		}
+
+		$transaction_isolation_level             = get_option( CustomOrdersTableController::DB_TRANSACTIONS_ISOLATION_LEVEL_OPTION, CustomOrdersTableController::DEFAULT_DB_TRANSACTIONS_ISOLATION_LEVEL );
+		$set_transaction_isolation_level_command = "SET TRANSACTION ISOLATION LEVEL $transaction_isolation_level";
```

Comment:
I think this is fine, given the context and given what MySQL even allows for this sort of statement, but I wanted to pause and ask anyway: is there any other escaping we might do here for safety (except for the fact we're using our own `db_query()` method, this would be normally have been flagged by the linter)?

Possibly comparing the string against a list of allowed values? (May not be necessary or even practical do to so, if which case we can move on.)